File: Arrays & Strings/6_pair_product.java
import java.util.List;
import java.util.HashMap;

class Source {
    public static List<Integer> pairProduct(List<Integer> numbers, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < numbers.size(); i++) {
            if (target % numbers.get(i) == 0) {
                int newTarget = target / numbers.get(i);
                if (map.containsKey(newTarget)) {
                    return List.of(map.get(newTarget), i);
                } else {
                    map.put(numbers.get(i), i);
                }
            }
        }

        return null;
    }
}

File: 1. Arrays & Strings/6_pair_product/6_pair_product.py
def pair_product(numbers, target):
    num_map = {}

    for i in range(len(numbers)):
        if target % numbers[i] == 0:
            new_target = target // numbers[i]
            if new_target in num_map:
                return [num_map[new_target], i]
            else:
                num_map[numbers[i]] = i
    
    return None
